xscen/aggregate.py

```diff
@@ -467,3 +491,282 @@ def spatial_mean(
+      the IPCC atlas data ( https://github.com/IPCC-WG1/Atlas/tree/main/datasets-aggregated-regionally/data )
+      and extra data from pilot models of MRCC5 and ClimEx.
+    to_level:
+      The processing level to assign to the output.
+      Use "{wl}" in the string to dynamically include the warming level.
+
+    Returns
+    -------
+    xr.Dataset
+        Warming level dataset
+    """
+    if min_window is None:
+        min_window = window - 2
+
+    if tas_csv is None:
+        tas_csv = Path(__file__).parent / "data/IPCC_annual_global_tas.csv"
+
+    # get info on ds
+    id_ds = ds.attrs["cat:id"]
+    source_ds = ds.attrs["cat:source"]
+    exp_ds = ds.attrs["cat:experiment"]
+    member_ds = ds.attrs["cat:member"]
+    mip_era_ds = ds.attrs["cat:mip_era"]
+
+    info_ds = (
+        f"{mip_era_ds}_{source_ds}_{exp_ds}_*"
+        if ignore_member
+        else f"{mip_era_ds}_{source_ds}_{exp_ds}_{member_ds}"
+    )
+
+    # open csv
+    annual_tas = pd.read_csv(tas_csv, index_col="year")
+
+    # choose colum based in ds cat attrs
+    right_column = annual_tas.filter(regex=info_ds, axis=1)
+
+    if len(right_column.columns) > 1:
+        logger.info(
+            "More than one column of the csv was selected. Choosing the first one."
+        )
+        right_column = pd.DataFrame(right_column.iloc[:, 0])
+    elif len(right_column.columns) == 0:
+        # TODO: should this fail ?
+        logger.info(
+            f"No columns fit the 'cat:' attributes of the input dataset ({info_ds}). Returning None."
+        )
+        return None
+
+    logger.info(
+        f"Computing warming level +{wl}C for id: {id_ds} from column: {right_column.columns[0]}."
+    )
+
+    # compute reference temperature for the warming
+    mean_base = right_column.loc[tas_base_period[0] : tas_base_period[1]].mean()
+
+    yearly_diff = right_column - mean_base  # difference from reference
+    last_year = right_column.iloc[-1].name  # last available year
+
+    # get the star and end date of the window when the warming level is first reached
+    for year in np.arange(1850, 2101):
+        start_yr = int(year - (window / 2 - 1))
+        end_yr = int(year + window / 2)
+        if end_yr > last_year:  # put nan if end_yr is not available in the data
+            start_yr = np.nan
+            end_yr = np.nan
+            break
+        rolling_mean = yearly_diff.loc[start_yr:end_yr].mean().values
+        if rolling_mean >= wl:
+            break
+        else:  # if it never goes above wl
+            start_yr = np.nan
+            end_yr = np.nan
+
+    if np.isnan(start_yr):
+        logger.info(
+            f"Global warming level of +{str(wl)}C is never reached for {id_ds}."
+        )
+        return None
+
+    # cut the window selected above
+    ds_wl = ds.sel(time=slice(str(start_yr), str(end_yr))).load()
+
+    # compute indicators
+    ind_dict = compute_indicators(ds=ds_wl, indicators=indicators)
```

Comment:
I think that everything below this is too specific. There are many use cases where you'd want to save and/or use the intermediate datasets, and not necessarily compute indicators or a climatological mean. This should be in a workflow, not a function?

Reply:
Not really sure what use cases you mean. To discuss thurday IRL.